pkg/logs/query.go

```diff
@@ -27,12 +27,13 @@ func Query(
 	q options.Query,
 	tls options.TLS,
 	defaultStep time.Duration,
-) (promapiv1.Warnings, error) {
+) (int, promapiv1.Warnings, error) {
 	// TODO: avoid type casting when we need to support all query endpoints for logs.
 	query, ok := q.(*options.QuerySpec)
 	if !ok {
-		return nil, errors.New("Incorrect query type for logs queries")
+		return http.StatusUnprocessableEntity, nil, errors.New("Incorrect query type for logs queries")
```

Comment:
```suggestion
		return 0 nil, errors.New("Incorrect query type for logs queries")
```

In line with the previous suggestions.